src/Log/Psr3Logger.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace ShoppinPal\YapepCommon\Log;
+
+use Psr\Log\LoggerInterface;
+use YapepBase\Config;
+use YapepBase\Log\LoggerAbstract;
+use YapepBase\Log\Message\IMessage;
+
+class Psr3Logger extends LoggerAbstract
+{
+
+    const APPLICATION_UNKNOWN = 'UNKNOWN';
+
+    /** @var string */
+    protected $programName;
+
+    /** @var string */
+    protected $applicationName;
+
+    /** @var LoggerInterface */
+    protected $logger;
+
+    /**
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+
+        $this->programName = Config::getInstance()->get('system.project.name');
+    }
+
+    /**
+     * @return LoggerInterface
+     */
+    public function getPsr3Logger()
```

Comment:
Why do you need this and why don't you just call it getLogger() ?

Reply:
This is currently not used, but it'd be useful for situations where you want to change the assigned handlers or something similar (say in testing). It's called getPsr3Logger to avoid confusion with the Yapep loggers.